### PR TITLE
Port from crossbeam-channel to flume

### PR DESCRIPTION
**Pull request by strohel**
This is a prerequisite for message priorities #22 - crossbeam has built-in
randomization when receiving messages, while flume lets us control that
using a feature flag.

flume's API is largely the same as crossbeam's, only real difference is
that select() is constructed using builder pattern rather than a macro.

Actor own benchmark result change (crossbeam -> flume):
```
circular/circular (2 actors)
                        time:   [3.0926 ms 3.1223 ms 3.1528 ms]
                        thrpt:  [317.18 Kelem/s 320.28 Kelem/s 323.35 Kelem/s]
                 change:
                        time:   [-24.762% -21.495% -17.992%] (p = 0.00 < 0.05)
                        thrpt:  [+21.940% +27.380% +32.912%]
                        Performance has improved.
Found 3 outliers among 100 measurements (3.00%)
  1 (1.00%) high mild
  2 (2.00%) high severe
circular/circular (native CPU count - 1)
                        time:   [3.9945 ms 4.0224 ms 4.0493 ms]
                        thrpt:  [246.96 Kelem/s 248.61 Kelem/s 250.34 Kelem/s]
                 change:
                        time:   [-7.5891% -5.5178% -3.4030%] (p = 0.00 < 0.05)
                        thrpt:  [+3.5229% +5.8400% +8.2123%]
                        Performance has improved.
Found 10 outliers among 100 measurements (10.00%)
  1 (1.00%) low severe
  6 (6.00%) low mild
  3 (3.00%) high mild
circular/circular (native CPU count)
                        time:   [20.821 ms 21.743 ms 22.666 ms]
                        thrpt:  [44.119 Kelem/s 45.993 Kelem/s 48.027 Kelem/s]
                 change:
                        time:   [-16.421% -10.076% -3.3897%] (p = 0.00 < 0.05)
                        thrpt:  [+3.5087% +11.205% +19.648%]
                        Performance has improved.
circular/circular (50 actors)
                        time:   [25.593 ms 27.095 ms 28.638 ms]
                        thrpt:  [34.918 Kelem/s 36.907 Kelem/s 39.073 Kelem/s]
                 change:
                        time:   [-31.525% -24.649% -17.014%] (p = 0.00 < 0.05)
                        thrpt:  [+20.502% +32.712% +46.038%]
                        Performance has improved.
```

That looks good! Measured on my laptop with frequency scaling and thermal
throttling, so there may be some noise.

### CODE REVIEW

File: src/lib.rs
Comment (skywhale) [inline src/lib.rs:87]: nit: `ControlChannel` is an internal concept, that the client of the library does not need to know about. You think it's okay to expose it as part of public API? or should we return `SystemStopped` since it's the only possible cause of `ControlChannelDisconnected`?

File: src/lib.rs
Comment (skywhale) [inline src/lib.rs:534]: nit: Did you struggle with the naming? There is something a bit confusing about it. There is `Recipient` already, but `Received` is not received by `Recipient` but by `Actor`. Not that I can think of a better name, but `ActorMessage`, `ActorSelectLoopMessage`, something along that line?

**Review by skywhale — APPROVED** [review]
Looks great :)

The throughput even increased?!

File: src/lib.rs
Comment (strohel) [reply src/lib.rs]: > Did you struggle with the naming?

I did. :) I was able to move `Received<M>` into `run_actor_select_loop()` - it is its implementation detail. Do you think that is sufficient to resolve the confusing naming, or shall we still rename it?

File: src/lib.rs
Comment (skywhale) [reply src/lib.rs:87]: I think that makes sense :) `unreachable!()/expect()` would be more self-documenting.

File: src/lib.rs
Comment (skywhale) [reply src/lib.rs]: That's sufficient. Thank you @strohel :)

File: src/lib.rs
Comment (strohel) [reply src/lib.rs]: Pushed the change. It resulted in more code cleanups, so I'll wait for another round of review before merging.

File: src/lib.rs
Comment (skywhale) [inline src/lib.rs:438]: typo: `should not happen`

Comment (skywhale) [reply src/lib.rs:438]: As per the suggestion from @mcginty on the other PR, let's switch to use `panic!()` here.

**Review by skywhale — APPROVED** [review]
LGTM++

File: src/lib.rs
Comment (strohel) [reply src/lib.rs]: :+1:

**Comment by bschwind** [comment]
Sorry I didn't review this earlier. Those performance numbers look great, and the replacement implementation seems pretty straightforward to me. Post-merge LGTM!

**Comment by strohel** [comment]
> Sorry I didn't review this earlier. Those performance numbers look great, and the replacement implementation seems pretty straightforward to me. Post-merge LGTM!

Thanks for post-merge review! :bow:

**Comment by mcginty** [comment]
Same as Brian! Looks great, and yay for easy performance gainzzz.

**Pull request merged**
